src/file_worker.py

```diff
@@ -335,6 +361,225 @@ def _get_dataset_scope(self, aDataset):
+    def _get_modification_time_of_dataset_files(self):
+
+        # run the query and morph results to an array of dict
+        with closing(self.hmdb.getDBConnection()) as dbConn:
+            with closing(dbConn.cursor(prepared=True)) as curs:
+                # query that finds all files associated with entity by joining the ancestors table (entity is
+                # the ancestor, files are the descendants) with the files table
+                curs.execute(SQL_SELECT_MOD_TIME_OF_DATASET_FILES)
+                results = [dict((curs.description[i][0].lower(), value) for i, value in enumerate(row)) for row in
+                           curs.fetchall()]
+
+        return results
+
+        # if len(results_json) < self.large_response_threshold:
+        #     return Response(response=results_json, mimetype="application/json")
+        # else:
+        #     return Response(self._stash_results_in_S3(object_content=results_json, key_uuid=uuid_tuple[0]), 303)
+
+    #
+    def _read_all_index_results(self, bearer_token):
+        scroll_id = None
+        current_read_hits = []
+        # Set up a dictionary which can be converted to JSON for opening a scroll search
+        # N.B. "scroll_open_minutes" is not a part of the OpenSearch JSON, but is how we signal
+        #      search-api's single scroll-search endpoint what to do.  In the future, if making
+        #      the search-api scroll-search public means more endpoints compatible with OpenSearch,
+        #      this method will be rewritten and can become more generic by passing in "query".
+        scroll_open_json_dict = {"scroll_open_minutes": self.max_minutes_open_scroll_context
+                                 ,"size": self.max_docs_per_scroll_page
+                                 ,"query": {"match_all": {}}
+                                 ,"fields": [ "file_uuid", "dataset_uuid", "file_info_refresh_timestamp"]
+                                 ,"_source": False
+                                 }
+        # Set up a dictionary which can be converted to JSON for continued reading of a scroll
+        scroll_read_json_dict = {"scroll_open_minutes": self.max_minutes_open_scroll_context
+                                    ,"scroll_id": "set after each response"}
+        # Set up a dictionary which can be converted to JSON for closing a scroll. Use zero minutes to
+        # indicate scroll should be closed.
+        scroll_dump_json_dict = {"scroll_open_minutes": 0
+                                    ,"scroll_id": "set after final response"}
+        # Open a scroll using the search-api composite index name 'files', expecting to
+        # get a scroll on the "consortium" (aka "private") OpenSearch index.  Assume the
+        # corresponding OpenSearch "public" index contains a subset of the "consortium" entries, each
+        # in the same state in both indices, as maintained by this method.
+        post_url = f"{self.search_api_url}/files/scroll-search"
```

Comment:
I think the generic nature of the string "files" might be clearer if some constant substituted it in.  My question would be, do we create confusion by having a configuration item that should never be changed? Maybe just a Python constant?

I could pop in a change for this after lunch, if you have a thought, @yuanzhou . Otherwise, maybe keep it in mind in the future?

Reply:
@kburke no problem, I don't have an opinion on this, either way it'll work fine. I've approved the PR.

Reply:
Thanks @yuanzhou!  @shirey, let me know if you want to get together on Zoom to walk through this and the whole state of files-api, esp. if you're thinking about architectural changes in the future.

Reply:
@kburke This is good for now, and we can definitely release, but add a card for a future sprint to make the /`files`/ index name configurable in files-api too.

Reply:
Done - https://github.com/hubmapconsortium/files-api/issues/26